vpn.tf

```diff
@@ -5,11 +5,11 @@ resource "azurerm_resource_group" "vpn" {
 }
 
 # Dedicated subnet in the private vnet
-resource "azurerm_subnet" "dmz" {
-  name                 = "${azurerm_virtual_network.private.name}-dmz"
+resource "azurerm_subnet" "vpn_internal" {
+  name                 = "${azurerm_virtual_network.private.name}-vpn-internal"
```

Comment:
```suggestion
resource "azurerm_subnet" "internal_vpn" {
  name                 = "${azurerm_virtual_network.private.name}-internalvpn"
```
Suggestion for the name, not mandatory.

If unwanted, I'd still remove the dash from the Azure name, to keep a coherency with the actual new resources names (nit):
```suggestion
resource "azurerm_subnet" "vpn_internal" {
  name                 = "${azurerm_virtual_network.private.name}-vpninternal"
```